Fix for #3976: add canonicalize_name() call for show

mkurnikov opened the pull request:
Fixes #3976.

xavfernandez commented on pip/commands/show.py:
```diff
@@ -4,6 +4,7 @@
+from pip._vendor.packaging.utils import canonicalize_name
 from pip.basecommand import Command
 from pip.status_codes import SUCCESS, ERROR
 from pip._vendor import pkg_resources
```
Please group the `from pip._vendor` imports together.

xavfernandez commented on tests/functional/test_show.py:
```diff
@@ -159,3 +159,14 @@ def test_pip_show_divider(script):
+
+@pytest.mark.network
+def test_package_name_is_canonicalized(script):
+    script.pip('install', 'py-salesforce')
```
I'd prefer if pip avoided a network call (tests are already flaky enough).
You could install https://github.com/pypa/pip/blob/master/tests/data/packages/pip-test-package-0.1.tar.gz instead
`script.pip('install', 'pip-test-package', '-f', data.packages)` should do it (with `def test_package_name_is_canonicalized(script, data)`)

mkurnikov commented:
Thanks for the review. I've addressed all comments in the latest commit. It looks like previous 
```
def test_pip_show_divider(script):
    ....
```
hitting the network too, so I marked it with decorator. 

BTW, please, create more `easy-fix` issues in the future.

xavfernandez commented on tests/functional/test_show.py:
```diff
@@ -151,6 +152,7 @@ def test_pip_show_is_short(script):
     assert len(lines) <= 10
 
 
+@pytest.mark.network
```
yup, sorry, you'll need to add `'--no-index'` to `script.pip('install', ...` to avoid any network call.

mkurnikov replied:
I've added it in `test_package_name_is_canonicalized`. Should I fix it for the `test_pip_show_divider` too?

xavfernandez replied:
Oh sorry, I did not spot it was on 2 separate tests.
Ideally, the change to `test_pip_show_divider` should not happen in this PR as it is unrelated...

mkurnikov replied:
Ok, I'll submit follow-up later then. 

Appveyor build has failed for absolutely unrelated reason, can I somehow restart only Appveyor, without Travis? I'd like not to wait 40 minutes for the new Travis build.

xavfernandez replied:
Again, this has nothing to do with this PR and #3976

pfmoore commented:
I've restarted the appveyor build

mkurnikov commented:
@pfmoore Thanks. 

@xavfernandez I think I've addressed everything you said, could it be merged?

xavfernandez commented on pip/commands/show.py:
```diff
@@ -1,12 +1,13 @@
-from email.parser import FeedParser
 import logging
 import os
+from email.parser import FeedParser
```
No need to move this

xavfernandez commented on pip/commands/show.py:
```diff
@@ -1,12 +1,13 @@
+from pip._vendor.packaging.utils import canonicalize_name
 from pip.basecommand import Command
 from pip.status_codes import SUCCESS, ERROR
-from pip._vendor import pkg_resources
```
No need to move this either :-/

xavfernandez commented:
@mkurnikov Could you rebase it on master ?

mkurnikov commented:
@xavfernandez Rebased. Also, I addressed all comments.

xavfernandez commented:
Thansk :+1:
